### PR TITLE
Jetpack Disconnect Survey: Final Touches

### PR DESCRIPTION
**Pull request by ockham**
* Fix punctuation by adding terminating periods to options.
* Simplify whitelist. Turns out we're free to use whatever slugs for reasons, so why not stick to the names we're using for routes and components. ~Note that I'm `camelCasing` them upon submission, for sake of consistency with the other marketing survey.~ Removed per discussion with @seear.

To test:
* `/settings/disconnect-site/:site`
* Verify that all options are terminated with a period (or 3 ellipsis dots)
* Verify that using the 'Back' button from the confirmation page still takes you back to the correct previous step.
* Verify that confirming submit the marketing survey with the reason camelCased (e.g. `too-difficult` become `tooDifficult`). (Check the 'Network' tab in your Dev Tools, Cmd+F `survey`)

### CODE REVIEW

**Comment by seear** [comment]
> Note that I'm camelCasing them upon submission, for sake of consistency with the other marketing survey

That might outfox someone grepping for the strings? Best to keep it simple and static?

File: client/my-sites/site-settings/disconnect-site/disconnect-survey.jsx [hunk 1]
```diff
@@ -28,20 +28,22 @@ const DisconnectSurvey = ( { confirmHref, isPaidPlan, siteId, siteSlug, translat
...
 		</CompactCard>
 		{ isPaidPlan && (
 			<CompactCard href={ `/settings/disconnect-site/too-expensive/${ siteSlug }` }>
-				{ translate( 'This plan is too expensive' ) }
+				{ translate( 'This plan is too expensive.' ) }
 			</CompactCard>
 		) }
 		<CompactCard href={ confirmHref + '?reason=troubleshooting' }>
-			{ translate( "Troubleshooting — I'll be reconnecting afterwards" ) }
+			{ translate( "Troubleshooting — I'll be reconnecting afterwards." ) }
```
Comment (a8ci18n) [inline client/my-sites/site-settings/disconnect-site/disconnect-survey.jsx]: :ok: This change is safe, we'll keep the existing 17 translations.

File: client/my-sites/site-settings/disconnect-site/disconnect-survey.jsx [hunk 2]
```diff
@@ -28,20 +28,22 @@ const DisconnectSurvey = ( { confirmHref, isPaidPlan, siteId, siteSlug, translat
...
-			{ translate( 'It was too hard to configure Jetpack' ) }
+			{ translate( 'It was too hard to configure Jetpack.' ) }
 		</CompactCard>
 		<CompactCard href={ `/settings/disconnect-site/missing-feature/${ siteSlug }` }>
 			{ translate( 'A feature I need was missing.' ) }
 		</CompactCard>
 		{ isPaidPlan && (
 			<CompactCard href={ `/settings/disconnect-site/too-expensive/${ siteSlug }` }>
-				{ translate( 'This plan is too expensive' ) }
+				{ translate( 'This plan is too expensive.' ) }
```
Comment (a8ci18n) [inline client/my-sites/site-settings/disconnect-site/disconnect-survey.jsx]: :ok: This change is safe, we'll keep the existing 18 translations.

File: client/my-sites/site-settings/disconnect-site/disconnect-survey.jsx [hunk 3]
```diff
@@ -28,20 +28,22 @@ const DisconnectSurvey = ( { confirmHref, isPaidPlan, siteId, siteSlug, translat
 			) }
 		/>
 		<CompactCard href={ `/settings/disconnect-site/too-difficult/${ siteSlug }` }>
-			{ translate( 'It was too hard to configure Jetpack' ) }
+			{ translate( 'It was too hard to configure Jetpack.' ) }
```
Comment (a8ci18n) [inline client/my-sites/site-settings/disconnect-site/disconnect-survey.jsx]: :ok: This change is safe, we'll keep the existing 17 translations.

**Comment by ockham** [comment]
>> Note that I'm camelCasing them upon submission, for sake of consistency with the other marketing survey

> That might outfox someone grepping for the strings? Best to keep it simple and static?

Yeah, can do. Wasn't too sure about this either.

**Comment by ockham** [comment]
`camelCasing` removed by 051f805.

**Pull request merged**